src/measure_theory/integral/set_to_l1.lean

```diff
@@ -0,0 +1,605 @@
+
+@[simp] lemma set_to_simple_func_zero {m : measurable_space α} (f : α →ₛ F) :
+  set_to_simple_func (0 : set α → F →L[ℝ] F') f = 0 :=
+by simp [set_to_simple_func]
+
+@[simp] lemma set_to_simple_func_zero_apply {m : measurable_space α} (T : set α → F →L[ℝ] F') :
+  set_to_simple_func T (0 : α →ₛ F) = 0 :=
+by casesI is_empty_or_nonempty α; simp [set_to_simple_func]
+
+lemma set_to_simple_func_eq_sum_filter {m : measurable_space α}
+  (T : set α → F →L[ℝ] F') (f : α →ₛ F) :
+  set_to_simple_func T f = ∑ x in f.range.filter (λ x, x ≠ 0), (T (f ⁻¹' {x})) x :=
+begin
+  symmetry,
+  refine sum_filter_of_ne (λ x hx, mt (λ hx0, _)),
+  rw hx0,
+  exact continuous_linear_map.map_zero _,
+end
+
+lemma set_to_simple_func_mono {G} [normed_linear_ordered_group G] [normed_space ℝ G]
+  {m : measurable_space α}
+  (T : set α → F →L[ℝ] G) (T' : set α → F →L[ℝ] G) (hTT' : ∀ s x, T s x ≤ T' s x) (f : α →ₛ F) :
+  set_to_simple_func T f ≤ set_to_simple_func T' f :=
+by { simp_rw set_to_simple_func, exact sum_le_sum (λ i hi, hTT' _ i), }
+
+lemma map_set_to_simple_func (T : set α → F →L[ℝ] F') (h_add : fin_meas_additive μ T)
+  {f : α →ₛ G} (hf : integrable f μ) {g : G → F} (hg : g 0 = 0) :
+  (f.map g).set_to_simple_func T = ∑ x in f.range, T (f ⁻¹' {x}) (g x) :=
+begin
+  have T_empty : T ∅ = 0, from map_empty_eq_zero_of_map_union T h_add,
+  have hfp : ∀ x ∈ f.range, x ≠ 0 → μ (f ⁻¹' {x}) ≠ ∞,
+    from λ x hx hx0, (measure_preimage_lt_top_of_integrable f hf hx0).ne,
+  simp only [set_to_simple_func, range_map],
+  refine finset.sum_image' _ (assume b hb, _),
+  rcases mem_range.1 hb with ⟨a, rfl⟩,
+  by_cases h0 : g (f a) = 0,
+  { simp_rw h0,
+    rw [continuous_linear_map.map_zero, finset.sum_eq_zero (λ x hx, _)],
+    rw mem_filter at hx,
+    rw [hx.2, continuous_linear_map.map_zero], },
+  have h_left_eq : T ((map g f) ⁻¹' {g (f a)}) (g (f a))
+    = T (f ⁻¹' ↑(f.range.filter (λ b, g b = g (f a)))) (g (f a)),
+  { congr, rw map_preimage_singleton, },
+  rw h_left_eq,
+  have h_left_eq' : T (f ⁻¹' ↑(filter (λ (b : G), g b = g (f a)) f.range)) (g (f a))
+    = T (⋃ y ∈ (filter (λ (b : G), g b = g (f a)) f.range), f ⁻¹' {y}) (g (f a)),
+  { congr, rw ← finset.set_bUnion_preimage_singleton, },
+  rw h_left_eq',
+  rw map_Union_fin_meas_set_eq_sum T T_empty h_add,
+  { simp only [filter_congr_decidable, sum_apply, continuous_linear_map.coe_sum'],
+    refine finset.sum_congr rfl (λ x hx, _),
+    rw mem_filter at hx,
+    rw hx.2, },
+  { exact λ i, measurable_set_fiber _ _, },
+  { intros i hi,
+    rw mem_filter at hi,
+    refine hfp i hi.1 (λ hi0, _),
+    rw [hi0, hg] at hi,
+    exact h0 hi.2.symm, },
+  { intros i j hi hj hij,
+    rw set.disjoint_iff,
+    intros x hx,
+    rw [set.mem_inter_iff, set.mem_preimage, set.mem_preimage, set.mem_singleton_iff,
+      set.mem_singleton_iff] at hx,
+    rw [← hx.1, ← hx.2] at hij,
+    exact absurd rfl hij, },
+end
+
+variables [normed_field 𝕜] [normed_space 𝕜 E] [smul_comm_class ℝ 𝕜 E]
```

Comment:
Is there a reason you are introducing this here and not when you need it later on?